test/functional/tests/cache_ops/test_concurrent_flushes.py

```diff
@@ -0,0 +1,191 @@
+        if int(core1.get_dirty_blocks()) > 0:
+            TestRun.LOGGER.error("The quantity of dirty cache lines on the first core "
+                                 "after complete flush should be zero.")
+
+        core2_dirty_blocks_after = int(core2.get_dirty_blocks())
+        if core2_dirty_blocks_after == 0:
+            TestRun.LOGGER.error("The quantity of dirty cache lines on the second core "
+                                 "after failed flush should not be zero.")
+        if core2_dirty_blocks_before != core2_dirty_blocks_after:
+            TestRun.LOGGER.error("The quantity of dirty cache lines on the second core "
+                                 "after failed flush should not change.")
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_concurrent_caches_flush(cache_mode):
+    """
+        title: Success to flush two caches simultaneously.
+        description: |
+          CAS should successfully flush multiple caches if there is already other flush in progress.
+        pass_criteria:
+          - No system crash.
+          - Every caches' flushes should finish successfully.
+    """
+    with TestRun.step("Prepare caches and cores."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([cache_size] * caches_number)
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([cache_size * 2] * caches_number)
+
+    with TestRun.step(f"Start {caches_number} caches."):
+        caches = []
+        for part in cache_dev.partitions:
+            caches.append(casadm.start_cache(part, cache_mode, force=True))
+
+    with TestRun.step("Disable cleaning and sequential cutoff policies."):
+        for cache in caches:
+            cache.set_cleaning_policy(CleaningPolicy.nop)
+            cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step(f"Add core devices to caches."):
+        cores = []
+        for i, cache in enumerate(caches):
+            cores.append(cache.add_core(core_dev.partitions[i]))
+
+    with TestRun.step("Run workload on every caches' core."):
+        # Every cache has one core fully saturated with dirty blocks.
+        bs = Size(4, Unit.MebiByte)
+        cnt = int(cache_size.value / bs.value)
+        for core in cores:
+            Dd().output(core.system_path) \
+                .input("/dev/urandom") \
+                .block_size(bs) \
+                .count(cnt) \
+                .run()
+
+    with TestRun.step("Check if all caches is full of dirty blocks."):
+        for cache in caches:
+            if int(cache.get_dirty_blocks()) == 0:
+                TestRun.fail(f"The cache {cache.cache_id} does not contain dirty blocks.")
```

Comment:
It is not a check if cache is full of dirty data. This 'if' checks whether cache is clean. Make statistic check according to test step message